gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2

```diff
@@ -1892,9 +1892,61 @@ def test_{{ method_name }}_empty_call():
 {% macro transport_kind_test(service, transport, is_async) %}
 {% set transport_name = get_transport_name(transport, is_async) %}
 def test_transport_kind_{{ transport_name }}():
+    {% if transport_name == 'rest_asyncio' %}
+    if not HAS_GOOGLE_AUTH_AIO:
+        {# TODO(https://github.com/googleapis/google-auth-library-python/pull/1577): Update the version of google-auth once the linked PR is merged. #}
+        pytest.skip("google-auth > 2.x.x is required for async rest transport.")
+    {% endif %}
     transport = {{ get_client(service, is_async) }}.get_transport_class("{{ transport_name }}")(
         credentials={{get_credentials(is_async)}}
     )
     assert transport.kind == "{{ transport_name }}"
 
 {% endmacro %}
+
+
+{% macro transport_close_test(service, transport, is_async) %}
+{% set async_prefix = "async " if is_async else "" %}
+{% set async_decorator = "@pytest.mark.asyncio " if is_async else "" %}
+{% set transport_name = get_transport_name(transport, is_async) %}
+{% set close_session = {
+    'rest': "_session",
+    'grpc': "_grpc_channel"}
+-%}
+{{async_decorator}}
+{{async_prefix}}def test_transport_close_{{transport_name}}():
+
+    {% if transport_name == 'rest_asyncio' %}
+    if not HAS_GOOGLE_AUTH_AIO:
+        {# TODO(https://github.com/googleapis/google-auth-library-python/pull/1577): Update the version of google-auth once the linked PR is merged. #}
+        pytest.skip("google-auth > 2.x.x is required for async rest transport.")
+    {% endif %}
+    client = {{ get_client(service, is_async) }}(
+        credentials={{get_credentials(is_async)}},
+        transport="{{transport_name}}"
+    )
+    with mock.patch.object(type(getattr(client.transport, "{{close_session[transport]}}")), "close") as close:
+        {{async_prefix}}with client:
+            close.assert_not_called()
+        close.assert_called_once()
+
+{% endmacro %}
+
+{# TODO(https://github.com/googleapis/gapic-generator-python/issues/2121): Remove / Update this test macro when async rest is GA. #}
+{% macro async_rest_unsupported_params_test(service) %}
+def test_unsupported_parameter_rest_asyncio():
+    if not HAS_GOOGLE_AUTH_AIO:
+        {# TODO(https://github.com/googleapis/google-auth-library-python/pull/1577): Update the version of google-auth once the linked PR is merged. #}
+        pytest.skip("google-auth > 2.x.x is required for async rest transport.")
+    options = client_options.ClientOptions(quota_project_id="octopus")
+    with pytest.raises(NotImplementedError) as exc:
+        client = {{ get_client(service, True) }}(
+            credentials={{get_credentials(True)}},
+            transport="rest_asyncio",
+            client_options=options
+        )
+    exc.match(
+        "The following provided parameters are not supported for `transport=rest_asyncio`: google.api_core.client_options.ClientOptions.quota_project_id"
+    )
```

Comment:
You could define and raise this in the appropriate module:
```
class AsyncRestUnsupportedParameterError(NotImplementedError):
    pass
```

and then test here a bit more robustly/succinctly:

```suggestion
    with pytest.raises(AsyncRestUnsupportedParameterError, match="google.api_core.client_options.ClientOptions.quota_project_id") :
        client = {{ get_client(service, True) }}(
            credentials={{get_credentials(True)}},
            transport="rest_asyncio",
            client_options=options
        )
    
```

Reply:
addressed here: https://github.com/googleapis/python-api-core/pull/694